src/librustc/ty/maps/mod.rs

```diff
@@ -227,6 +227,8 @@ define_maps! { <'tcx>
     [] fn item_body_nested_bodies: ItemBodyNestedBodies(DefId) -> ExternBodyNestedBodies,
     [] fn const_is_rvalue_promotable_to_static: ConstIsRvaluePromotableToStatic(DefId) -> bool,
     [] fn is_mir_available: IsMirAvailable(DefId) -> bool,
+    [] fn vtable_methods: vtable_methods_node(ty::PolyTraitRef<'tcx>)
+                          -> Vec<Option<(DefId, &'tcx Substs<'tcx>)>>,
```

Comment:
Returning `Rc<Vec<Option<..>>>` would be better.

Reply:
Right! I had that change before, I should re-apply that patch.